python/lsst/ctrl/mpexec/separablePipelineExecutor.py

```diff
@@ -0,0 +1,83 @@
+# This file is part of ctrl_mpexec.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+__all__ = [
+    "SeparablePipelineExecutor",
+]
+
+
+from typing import Optional
+
+import lsst.pipe.base
+from lsst.daf.butler import Butler
+
+from .taskFactory import TaskFactory
+
+
+class SeparablePipelineExecutor:
+    """An executor that allows each step of pipeline execution to be
+    run independently.
+
+    The executor can run any or all of the following steps:
+
+        * pre-execution initialization
+        * pipeline building
+        * quantum graph generation
+        * quantum graph execution
+
+    Any of these steps can also be handed off to external code without
+    compromising the remaining ones.
+
+    Parameters
+    ----------
+    butler : `lsst.daf.butler.Butler`
+        A Butler whose ``collections`` and ``run`` attributes contain the input
+        and output collections to use for processing.
+    clobber_output : `bool`, optional
+        If set, the pipeline execution overwrites existing output files.
+        Otherwise, any conflict between existing and new outputs is an error.
+    skip_existing : `bool`, optional
+        If set, the pipeline execution skips any tasks that already have all
+        their outputs. Otherwise, all tasks are attempted (subject
```

Comment:
Well, the "other docs" I used as a reference are `pipetask run --help`... 😬 

> But on that note, why not accept skip_existing_in as an iterable of some kind instead? Is that hard to support somewhere?

No, it would actually be useful for us. It just wasn't as intuitive an API for me. I'll change it.

Reply:
> But on that note, why not accept skip_existing_in as an iterable of some kind instead? Is that hard to support somewhere?

No, it would actually be useful for us. It just wasn't as intuitive an API for me. I'll change it.

Reply:
> Is there another way of wording it that accounts for optional outputs but keeps the spirit of "task ran fully to completion"?

"If a previous execution of an equivalent quantum ran to completion or identified that it had no work to do."